libnetwork/netutils/utils.go

```diff
@@ -62,35 +62,6 @@ func NetworkRange(network *net.IPNet) (net.IP, net.IP) {
 	return firstIP, lastIP
 }
 
-// GetIfaceAddr returns the first IPv4 address and slice of IPv6 addresses for the specified network interface
-func GetIfaceAddr(name string) (net.Addr, []net.Addr, error) {
-	iface, err := net.InterfaceByName(name)
-	if err != nil {
-		return nil, nil, err
-	}
-	addrs, err := iface.Addrs()
```

Comment:
Just a random comment 🤔 

Haven't been through all changes yet, but I saw this being removed when I looked from my phone, and I was curious why it wasn't used (as some options accept either an IP-address or an interface name). But it looks like we have various "variations" on a a theme (some are test-utilities, so perhaps irrelevant). I haven't compared them fully (to see if there's specific differences between them), but was wondering if there would be use for a common implementation (or if it would be worth looking into);

- getBindAddr https://github.com/moby/moby/blob/dd3b71d17c614f837c4bba18baed9fa2cb31f1a4/libnetwork/agent.go#L58-L62
- resolveInterfaceAddr https://github.com/moby/moby/blob/dd3b71d17c614f837c4bba18baed9fa2cb31f1a4/daemon/cluster/listen_addr.go#L161-L168
- getExternalAddress https://github.com/moby/moby/blob/dd3b71d17c614f837c4bba18baed9fa2cb31f1a4/integration/container/nat_test.go#L121-L126
- setupDriver https://github.com/moby/moby/blob/dd3b71d17c614f837c4bba18baed9fa2cb31f1a4/libnetwork/drivers/overlay/overlay_test.go#L66-L70

Reply:
`libnetwork.getBindAddr` returns an arbitrary interface IP address which is not an IPv6 link-local address, with no preference of IP address family. On the other hand, `cluster.resolveInterfaceAddr` prefers IPv4 over IPv6, neglects to skip over IPv6 link-local addresses, and returns an error if there is more than one address of either family associated with the interface. Since IPv6 requires a link-local address be assigned to each interface in addition to any routable addresses, `cluster.resolveInterfaceAddr` is broken for any interface which has a routable IPv6 address associated.

I question the utility of both functions. While it is kinda convenient to let the administrator specify an interface name instead of an address, specifying an interface implies a promise which the daemon can't currently keep: that the daemon will dynamically handle changes to the IP addresses associated with the interface.

Reply:
>  question the utility of both functions. While it is kinda convenient to let the administrator specify an interface name instead of an address, specifying an interface implies a promise which the daemon can't currently keep: that the daemon will dynamically handle changes to the IP addresses associated with the interface.

Yeah, swarm doesn't handle IP changes at all. The "allow interface" was mostly out of convenience (ISTR part of that was created at the time for automated setups of clusters)

It still feels like a good thing to verify the various code paths to check if they're consistent (and do the right thing).

Definitely not for this PR though!